critiquebrainz/data/dump_manager.py

```diff
@@ -364,21 +367,18 @@ def create_reviews_archive(*, location, meta_files_dir=None, license_id=None):
         temp_dir = tempfile.mkdtemp()
         reviews_tables_dir = os.path.join(temp_dir, safe_name)
         create_path(reviews_tables_dir)
-        with db.engine.connect() as connection:
-            with connection.begin() as transaction:
-                cursor = connection.connection.cursor()
-                try:
-                    with open(os.path.join(reviews_tables_dir, 'review'), 'w') as f:
-                        cursor.copy_to(f, REVIEW_SQL)
-
-                    with open(os.path.join(reviews_tables_dir, 'revision'), 'w') as f:
-                        cursor.copy_to(f, REVISION_SQL)
-
-                    with open(os.path.join(reviews_tables_dir, 'avg_rating'), 'w') as f:
-                        cursor.copy_to(f, "(SELECT {columns} FROM avg_rating)".format(columns=", ".join(_TABLES["avg_rating"])))
-                except Exception:
-                    print("Error while copying tables during the creation of the reviews archive")
-                    transaction.rollback()
+        cursor = connection.connection.cursor()
+        try:
+            with open(os.path.join(reviews_tables_dir, 'review'), 'w') as f:
+                cursor.copy_to(f, REVIEW_SQL)
+
+            with open(os.path.join(reviews_tables_dir, 'revision'), 'w') as f:
+                cursor.copy_to(f, REVISION_SQL)
+
+            with open(os.path.join(reviews_tables_dir, 'avg_rating'), 'w') as f:
+                cursor.copy_to(f, "(SELECT {columns} FROM avg_rating)".format(columns=", ".join(_TABLES["avg_rating"])))
+        except Exception:
+            print("Error while copying tables during the creation of the reviews archive")
```

Comment:
If you're catching the exception here, you'll have to rollback the transaction manually because otherwise it'll move out of the with block with no error and commit the transaction

Or just raise the exception back

Reply:
@paramsingh I've checked this manually. Whether you catch the exception or not, `__exit__()` function will still be called. Since here, the `__exit__()` function itself contains the `transaction.rollback()` we don't need to call it explicitly.

Reply:
Why would it call the `transaction.rollback` if no exception is raised and the exit is a normal exit? 



```
with db.engine.begin() as transaction:
     pass
```
would probably do a transaction.commit

```
with db.engine.begin() as transaction:
     raise Exception
```
should rollback

the code where we're catching the exception and just logging it is equivalent to the first case?

Reply:
@paramsingh Sorry, my bad. Now I get it. Let's just raise the exception in that case, rather that explicitly writing rollback. What you say?

Reply:
Right, okay. We should just be careful that the Exception reaches the toplevel `with` block for sure.

Reply:
Decided to use nested transactions instead.